src/components/button/button.test.tsx

```diff
@@ -0,0 +1,330 @@
+    render(<Button aria-label="foo">bar</Button>);
+
+    const button = screen.getByRole("button");
+
+    expect(button).toHaveAttribute("aria-label", "foo");
+  });
+
+  it("sets the 'aria-labelledby' attribute correctly when a custom value is passed to the 'aria-labelledby' prop", () => {
+    render(<Button aria-labelledby="foo">bar</Button>);
+
+    const button = screen.getByRole("button");
+
+    expect(button).toHaveAttribute("aria-labelledby", "foo");
+  });
+
+  it("sets the 'aria-describedby' attribute correctly when a custom value is passed to the 'aria-describedby' prop", () => {
+    render(<Button aria-describedby="foo">bar</Button>);
+
+    const button = screen.getByRole("button");
+
+    expect(button).toHaveAttribute("aria-describedby", "foo");
+  });
+
+  /* Styling test for coverage */
+  it("sets the correct icon colour when 'gradient-white' is passed to the 'buttonType' prop", () => {
+    render(
+      <Button aria-label="button" buttonType="gradient-white" iconType="bin">
+        foo
+      </Button>
+    );
+
+    const icon = screen.getByTestId("icon");
+    expect(icon).toHaveStyle("color: --colorsActionMinorYin090");
+  });
+
+  /* Styling test for coverage */
+  it("sets the correct icon colour when 'gradient-grey' is passed to the 'buttonType' prop", () => {
+    render(
+      <Button aria-label="button" buttonType="gradient-grey" iconType="bin">
+        foo
+      </Button>
+    );
+
+    const icon = screen.getByTestId("icon");
+    expect(icon).toHaveStyle("color: --colorsActionMinorYin090");
+  });
+
+  /* Styling test for coverage */
+  it("sets the correct height when 'services' is passed to the 'buttonType' prop", () => {
+    render(
+      <Button aria-label="button" iconType="services" size="large">
+        foo
+      </Button>
+    );
+
+    const icon = screen.getByTestId("icon");
+    expect(icon).toHaveStyle("height: 6px");
+  });
+
+  it("sets the 'name' attribute correctly when a custom value is passed to the 'name' prop", () => {
+    render(<Button name="foo">bar</Button>);
+
+    const button = screen.getByRole("button");
+
+    expect(button).toHaveAttribute("name", "foo");
+  });
+
+  it("sets the 'disabled' attribute correctly when the 'disabled' prop is set to 'true'", () => {
+    render(<Button disabled>bar</Button>);
+
+    const button = screen.getByRole("button");
+
+    expect(button).toBeDisabled();
+  });
+
+  /* Styling test for coverage */
+  it("sets the correct styling attributes when the 'destructive' and 'disabled' props are 'true'", () => {
+    render(
+      <Button destructive disabled>
+        foo
+      </Button>
+    );
+
+    const button = screen.getByRole("button");
+
+    expect(button).toHaveStyleRule("color: var(--colorsActionMajorYin030)");
+    expect(button).toHaveStyleRule(
+      "background-color: var(--colorsActionMajorYin030)"
+    );
+    expect(button).toHaveStyleRule(
+      "background: var(--colorsActionDisabled500)"
+    );
```

Comment:
**question(non-blocking):** Did you try these as `toHaveStyle` instead? Just thinking that if `toHaveStyle` works, you'd be able to assert these styles in one block as `toHaveStyle` allows an object.

Reply:
I believe this was a test case where `toHaveStyle` was returning some weird original styling, whereas `toHaveStyleRule` passed, and we need the test for coverage